drivers/gpu/drm/vc4/vc4_kms.c

```diff
@@ -427,7 +427,7 @@ static void vc4_atomic_commit_tail(struct drm_atomic_state *state)
 
 	drm_atomic_helper_cleanup_planes(dev, state);
 
-	if (vc4->hvs && vc4->hvs->hvs5) {
+	if (vc4->is_vc5) {
```

Comment:
This breaks fkms (where `vc4->hvs` would be null). Suggest `if (vc4->is_vc5 && !vc4->firmware_kms)` as fix.